src/test/java/de/retest/recheck/ignore/SearchFilterFilesTest.java

```diff
@@ -135,6 +136,25 @@ void getFilterByName_should_return_user_filter_with_given_name_over_default_filt
 		assertThat( result.toString() ).isEqualTo( "CompoundFilter(filters=[])" );
 	}
 
+	@Test
+	@ClearSystemProperty( key = RETEST_PROJECT_ROOT )
+	void getFilterByName_should_only_load_searched_filter() throws IOException {
+		System.setProperty( RETEST_PROJECT_ROOT, filterFolder.toString() );
+
+		// Create infinite filter loop which will cause infinite loop if loaded
+		final Path resolve = filterFolder.resolve( "foo.filter" );
+		Files.write( resolve, Collections.singleton( "import: bar.filter" ) );
+		final Path bar = filterFolder.resolve( "bar.filter" );
+		Files.write( resolve, Collections.singleton( "import: foo.filter" ) );
+
+		final Path positioningFilter = filterFolder.resolve( "content.filter" );
+		Files.createFile( positioningFilter );
+
+		final Filter result = SearchFilterFiles.getFilterByName( "content.filter" );
+		assertThat( result ).isNotNull();
+		assertThat( result.toString() ).isEqualTo( "CompoundFilter(filters=[])" );
```

Comment:
Rather than asserting the string representation, AssertJ has:

```java
assertThat( result ).isInstanceOfSatisfying( CompoundFilter.class,
		compound -> assertThat( compound.getFilters() ).isEmpty() );
```